File: src/config/ConfigSettings.cs
using System;
using System.IO;
using Newtonsoft.Json;

public class ConfigSettings
{
    // Visualization Parameters
    public string BackgroundColor { get; set; }
    public string NodeColor { get; set; }
    public string EdgeColor { get; set; }
    public int NodeSize { get; set; }
    public int EdgeWidth { get; set; }

    // File Paths
    public string DataFile { get; set; }
    public string LogFile { get; set; }

    // UI Constants
    public int WindowWidth { get; set; }
    public int WindowHeight { get; set; }

    public static ConfigSettings LoadSettings(string filePath)
    {
        if (File.Exists(filePath))
        {
            string json = File.ReadAllText(filePath);
            return JsonConvert.DeserializeObject<ConfigSettings>(json);
        }
        else
        {
            var defaultSettings = new ConfigSettings
            {
                BackgroundColor = ConfigConstants.BackgroundColor,
                NodeColor = ConfigConstants.NodeColor,
                EdgeColor = ConfigConstants.EdgeColor,
                NodeSize = ConfigConstants.NodeSize,
                EdgeWidth = ConfigConstants.EdgeWidth,
                DataFile = ConfigConstants.DefaultDataFile,
                LogFile = ConfigConstants.LogFile,
                WindowWidth = ConfigConstants.WindowWidth,
                WindowHeight = ConfigConstants.WindowHeight
            };
            SaveSettings(defaultSettings, filePath);
            return defaultSettings;
        }
    }

    public static void SaveSettings(ConfigSettings settings, string filePath)
    {
        string json = JsonConvert.SerializeObject(settings, Formatting.Indented);
        File.WriteAllText(filePath, json);
    }
}

public class VisualizationParams
{
    public string BackgroundColor { get; set; }
    public string NodeColor { get; set; }
    public string EdgeColor { get; set; }
    public int NodeSize { get; set; }
    public int EdgeWidth { get; set; }
}

File: src/config/ConfigSettings.cpp
#include "ConfigSettings.hpp"
#include <godot_cpp/classes/file_access.hpp>
#include <godot_cpp/classes/json.hpp>

namespace Config {

void ConfigSettings::_bind_methods() {
    ClassDB::bind_method(D_METHOD("set_background_color", "color"), &ConfigSettings::set_background_color);
    ClassDB::bind_method(D_METHOD("get_background_color"), &ConfigSettings::get_background_color);
    // Bind other properties...

    ADD_PROPERTY(PropertyInfo(Variant::STRING, "background_color"), "set_background_color", "get_background_color");
    // Add other properties...
}

ConfigSettings::ConfigSettings() {
    // Set defaults from Constants
    background_color = Constants::BACKGROUND_COLOR;
    node_color = Constants::NODE_COLOR;
    edge_color = Constants::EDGE_COLOR;
    node_size = Constants::NODE_SIZE;
    edge_width = Constants::EDGE_WIDTH;
    data_file = Constants::DEFAULT_DATA_FILE;
    log_file = Constants::LOG_FILE;
    window_width = Constants::WINDOW_WIDTH;
    window_height = Constants::WINDOW_HEIGHT;
}

Ref<ConfigSettings> ConfigSettings::load_settings(const String& file_path) {
    Ref<ConfigSettings> settings;
    settings.instantiate();

    Ref<FileAccess> f = FileAccess::open(file_path, FileAccess::READ);
    if (f.is_valid()) {
        String json_str = f->get_as_text();
        Variant parsed = JSON::parse_string(json_str);
        
        if (parsed.get_type() == Variant::DICTIONARY) {
            Dictionary data = parsed;
            // Load values from dictionary
            settings->background_color = data.get("background_color", Constants::BACKGROUND_COLOR);
            // Load other properties...
        }
    }
    
    return settings;
}

void ConfigSettings::save_settings(const String& file_path) {
    Dictionary data;
    data["background_color"] = background_color;
    // Add other properties to dictionary...

    String json_str = JSON::stringify(data);
    
    Ref<FileAccess> f = FileAccess::open(file_path, FileAccess::WRITE);
    if (f.is_valid()) {
        f->store_string(json_str);
    }
}

} 